Moblin/RtmpServer/RtmpServerChunkStream.swift

```diff
@@ -471,40 +445,31 @@ class RtmpServerChunkStream {
             logger.info("rtmp-server: client: Dropping short packet with data \(messageData.hexString())")
             return
         }
-        if let sampleBuffer = makeSampleBuffer(client: client) {
+        if let sampleBuffer = makeVideoSampleBuffer(client: client) {
             videoDecoder?.appendSampleBuffer(sampleBuffer)
         } else {
             client.stopInternal(reason: "Make sample buffer failed")
         }
     }
 
-    private func makeSampleBuffer(client: RtmpServerClient) -> CMSampleBuffer? {
-        var compositionTime = Int32(data: [0] + messageData[2 ..< 5]).bigEndian
-        compositionTime <<= 8
-        compositionTime /= 256
-        var duration = Int64(messageTimestamp)
-        if isMessageType0 {
-            if videoTimestampZero == -1 {
-                videoTimestampZero = Double(messageTimestamp)
-            }
-            duration -= Int64(videoTimestamp)
-            videoTimestamp = Double(messageTimestamp) - videoTimestampZero
-        } else {
-            videoTimestamp += Double(messageTimestamp)
-        }
-        var presentationTimeStamp: Int64
-        var decodeTimeStamp: Int64
-        if client.fps == 0 {
-            presentationTimeStamp = Int64(videoTimestamp) + Int64(compositionTime)
-            decodeTimeStamp = Int64(videoTimestamp)
```

Comment:
Not handling PTS and DTS separetly will probably break video streams with B-frames. That is probably fine, as I'm sure very few, if any, of the RTMP publishers that feeds data to Mobli uses B-frames.

Reply:
I think b frames would be handy to have, especially given the quality benefit. Seems worthwhile to have, and separating PTS/DTS isn't really that complicated to implement.